packages/testing/src/api/mockWidgetApi.test.ts

```diff
@@ -822,3 +822,125 @@ describe('readEventRelations', () => {
     ).rejects.toThrow('Unexpected error while reading relations');
   });
 });
+
+describe('sendToDeviceMessage', () => {
+  it('should send to device message to all devices of the current user', async () => {
+    const messagePromise = firstValueFrom(
+      widgetApi.observeToDeviceMessages('com.example.message')
+    );
+
+    await widgetApi.sendToDeviceMessage('com.example.message', true, {
+      '@user-id': {
+        '*': {
+          my: 'content',
+        },
+      },
+    });
+
+    await expect(messagePromise).resolves.toEqual({
+      content: {
+        my: 'content',
+      },
+      encrypted: true,
+      sender: '@user-id',
+      type: 'com.example.message',
+    });
+  });
+
+  it('should send to device message to a specific device of the current user', async () => {
+    const messagePromise = firstValueFrom(
+      widgetApi.observeToDeviceMessages('com.example.message')
+    );
+
+    await widgetApi.sendToDeviceMessage('com.example.message', false, {
+      '@user-id': {
+        'device-id': {
+          my: 'content',
+        },
+      },
+    });
+
+    await expect(messagePromise).resolves.toEqual({
+      content: {
+        my: 'content',
+      },
+      encrypted: false,
+      sender: '@user-id',
+      type: 'com.example.message',
+    });
+  });
+
+  it('should send to device message to another user', async () => {
+    const messagesPromise = firstValueFrom(
+      widgetApi
+        .observeToDeviceMessages('com.example.message')
+        .pipe(bufferTime(100))
+    );
+
+    await widgetApi.sendToDeviceMessage('com.example.message', false, {
+      '@other-user-id': {
+        '*': {
+          other: 'content',
+        },
+      },
+    });
+
+    await expect(messagesPromise).resolves.toEqual([]);
+  });
+
+  it('should send to device message to multiple users', async () => {
+    const messagesPromise = firstValueFrom(
+      widgetApi
+        .observeToDeviceMessages('com.example.message')
+        .pipe(bufferTime(100))
+    );
+
+    await widgetApi.sendToDeviceMessage('com.example.message', false, {
+      '@other-user-id': {
+        '*': {
+          other: 'content',
+        },
+      },
+      '@user-id': {
+        '*': {
+          my: 'content',
+        },
+      },
+    });
+
+    await expect(messagesPromise).resolves.toEqual([
+      {
+        content: { my: 'content' },
+        sender: '@user-id',
+        encrypted: false,
+        type: 'com.example.message',
+      },
+    ]);
+  });
+});
+
+describe('observeToDeviceMessages', () => {
+  it('should receive only to device messages for the correct type', async () => {
+    const messagePromise = firstValueFrom(
+      widgetApi.observeToDeviceMessages('com.example.message')
+    );
+
+    widgetApi.mockSendToDeviceMessage({
```

Comment:
Would it make sense to send a second device message of another type first so we know that is correctly skipped?

Reply:
Total correct!